bolt/lib/Core/DIEBuilder.cpp

```diff
@@ -77,6 +77,9 @@ static void addStringHelper(DebugStrOffsetsWriter &StrOffstsWriter,
                             DIEValue &DIEAttrInfo, StringRef Str) {
   uint32_t NewOffset = StrWriter.addString(Str);
   if (Unit.getVersion() >= 5) {
+    assert(StrOffstsWriter.getDebugStrOffsetsFinalized().count(
```

Comment:
Move this inside updateAddressMap. There is no guarantee that in the future it might not be called from somewhere else.